### PR TITLE
clinfo: init at 2.2.18.04.06

### PR DESCRIPTION
**Pull request by athas**
###### Motivation for this change

`clinfo` is the standard tool for querying the state of OpenCL.  Nixpkgs already has `opencl-info`, but it's older and unmaintained.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
Aw man I was looking for this the other day, and was unhappy to find opencl-info seemed the best available.  Thanks for the good timing! ;).

File: pkgs/tools/system/clinfo/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, ocl-icd, opencl-headers }:
+
+stdenv.mkDerivation rec {
+  pname = "clinfo";
+  version = "2.2.18.04.06";
+
+  src = fetchurl {
```
Comment (jtojnar) [inline pkgs/tools/system/clinfo/default.nix]: You should use `fetchFromGitHub` here.

File: pkgs/tools/system/clinfo/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://github.com/Oblomov/${pname}/archive/${version}.tar.gz";
+    sha256 = "f77021a57b3afcdebc73107e2254b95780026a9df9aa4f8db6aff11c03f0ec6c";
+  };
+
+  buildInputs = [ ocl-icd opencl-headers ];
+
+  NIX_LDFLAGS = [ "-lOpenCL" ];
+
+  installPhase = ''
```
Comment (jtojnar) [inline pkgs/tools/system/clinfo/default.nix]: `makeFlags = [ "PREFIX=${placeholder "out"}" ];` should allow you to get rid of the custom `installPhase`.

File: pkgs/tools/system/clinfo/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  version = "2.2.18.04.06";
+
+  src = fetchurl {
+    url = "https://github.com/Oblomov/${pname}/archive/${version}.tar.gz";
+    sha256 = "f77021a57b3afcdebc73107e2254b95780026a9df9aa4f8db6aff11c03f0ec6c";
+  };
+
+  buildInputs = [ ocl-icd opencl-headers ];
+
+  NIX_LDFLAGS = [ "-lOpenCL" ];
```
Comment (jtojnar) [inline pkgs/tools/system/clinfo/default.nix]: Why is this necessary? They appear to already be setting it https://github.com/Oblomov/clinfo/blob/9d3f66f603d1e76173ed42be1d9181c991fb4226/Makefile#L43

**Comment by athas** [comment]
Thanks for the review; I have fixed everything that was pointed out.

**Review by jtojnar — APPROVED** [review]

File: pkgs/tools/system/clinfo/default.nix (hunk 2 above)
Comment (gloaming) [reply pkgs/tools/system/clinfo/default.nix]: You can also just pass PREFIX=$(out) in this case, and Make will expand the environment variable $out.
